src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs

```diff
@@ -1085,18 +1100,12 @@ private bool AwaitingLease(object message)
         {
             switch (message)
             {
-                case LeaseAcquireResult lar when lar.Acquired:
+                case LeaseAcquireResult { Acquired: true }:
                     Log.Debug("{0}: Lease acquired", _typeName);
                     TryLoadRememberedEntities();
                     return true;
 
-                case LeaseAcquireResult lar when !lar.Acquired && lar.Reason == null:
```

Comment:
Code removed and simplified, `ILoggingAdaptor.Error` can accept null exception, this code is not needed.